GitUI/CommandsDialogs/BrowseDialog/FormCommitCount.cs

```diff
@@ -24,9 +24,11 @@ public FormCommitCount(GitUICommands commands)
 
             Load += delegate { FetchData(); };
             cbIncludeSubmodules.CheckedChanged += delegate { FetchData(); };
+
+            void FetchData() => ThreadHelper.JoinableTaskFactory.RunAsync(FetchDataAsync).FileAndForget();
         }
 
-        private void FetchData()
+        private async Task FetchDataAsync()
         {
             ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
:memo: Since this method is `private`, it's not too difficult to know that this condition holds. However, in general cases you'd replace this like with `await this.SwitchToMainThreadAsync()`.

Reply:
I feel like a throw is fine here, given the conditions are knowable at-a-glance.

I'd probably even go for `ThreadHelper.AssertOnUIThread()` with `[Conditional("DEBUG")]` in this case as it's really an error on the part of the developer if you're not on the UI thread here.